src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Marshaling/JSMarshalerArgument.Task.cs

```diff
@@ -49,7 +49,7 @@ public unsafe void ToManaged(out Task? value)
             lock (ctx)
             {
                 PromiseHolder holder = ctx.GetPromiseHolder(slot.GCHandle);
-                TaskCompletionSource tcs = new TaskCompletionSource(holder);
+                TaskCompletionSource tcs = new TaskCompletionSource(holder, TaskCreationOptions.DenyChildAttach | TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Could please put comments with reasoning about the task options used?

Reply:
I mixed in some half baked experiment around that. And I do struggle with that, so I will fix it and comment it, so that you can validate my thinking about it.

Reply:
I gave up today because my machine is not in good shape and this needs to be merged over weekend. 
I only put in some TODO in those places.